src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -549,59 +549,43 @@ export function useSwapInputsController({
     { leading: false, trailing: true }
   );
 
+  const resetValuesToZeroWorklet = (inputKey?: inputKeys) => {
+    'worklet';
+    quoteFetchingInterval.stop();
+    if (isFetching.value) isFetching.value = false;
+    if (isQuoteStale.value !== 0) isQuoteStale.value = 0;
+    if (inputKey) lastTypedInput.value = inputKey;
+
+    const keysToReset = ['inputAmount', 'inputNativeValue', 'outputAmount', 'outputNativeValue'];
+    const inputKeyValue = inputKey ? inputValues.value[inputKey] : 0;
+    const hasDecimal = inputKeyValue.toString().includes('.');
+    const updatedValues = keysToReset.reduce(
+      (acc, key) => {
+        const castedKey = key as keyof typeof inputValues.value;
+        acc[castedKey] = castedKey === inputKey && hasDecimal ? inputValues.value[castedKey] : 0;
+        return acc;
+      },
+      {} as Partial<typeof inputValues.value>
+    );
+
+    inputValues.modify(values => {
+      return {
+        ...values,
+        ...updatedValues,
+      };
+    });
+
+    if (inputKey) {
+      sliderXPosition.value = withSpring(0, snappySpringConfig);
+    }
+  };
+
   const onTypedNumber = useDebouncedCallback(
-    (amount: number, inputKey: inputKeys, preserveAmount = true) => {
+    (amount: number, inputKey: inputKeys) => {
       lastTypedInput.value = inputKey;
 
       if (amount > 0) {
-        const updateWorklet = () => {
-          'worklet';
-          // If the user enters a new inputAmount, update the slider position ahead of the quote fetch, because
-          // we can derive the slider position directly from the entered amount.
-          if (inputKey === 'inputAmount') {
```

Comment:
This code was only used for when the `inputKey` is the `inputAmount` and is already done on lines 741-752 before the function is called, so it is redundant.